### ISSUE TITLE
Enhanced register process

### ISSUE THREAD

**Issue by lauraroverts** [akvo/akvo-rsr#1611]
Registration process is quite cumbersome. It takes a lot of actions before a user can actually start creating updates. First you need to register yourself, and afterwards you need to connect to your organisation. It would be much easier if you can do that immediately. So, for instance making the registration page exist out of 2 pages: 1) personal registration (password, email etc) 2) connect to employer. Then users need to go only once to their e-mail to activate everything. 

Also, in MyRSR, when users click the activation link in the email, they get redirected to a message saying you need to go to myRSR.
![screenshot 1](https://cloud.githubusercontent.com/assets/4179989/8006726/8dfe836c-0b96-11e5-8f76-fcd3f930173c.png)

It is not obvious that the next step is to connect with their employer.
![screenshot 2 copy](https://cloud.githubusercontent.com/assets/4179989/8006764/edc2d320-0b96-11e5-946b-aaa1b08f1dde.png)

If we keep to the same process as it is now, it should become more clear what they need to do after clicking the activation link. Maybe an automated pop-up in MyRSR that directs them to "Connect with your employer".

**Comment by lauraroverts** [comment]
See also https://github.com/akvo/akvo-product-design/issues/122

**Comment by KasperBrandt** [comment]
The biggest issue in this case is the two-step registration process: first create your user account, and then link to an organisation. As a first step to improve this process, we should add an organisation select box to the register page. The process will then be as follows:

1. User fills in personal information and organisation (organisation will be required from now on).
2. User account (inactive) and user employment record (not approved) are created.
3. Email is sent to user to activate their account.
4. Upon activation, an email is sent to the admins of this organisation and the organisation that is the content_owner of the organisation (and the content_owner of that organisation, etc), who should be able to approve the user in the User Management page. If possible, we could add 'Approve' and 'Decline' buttons already in this email.

**Comment by lauraroverts** [comment]
In addition regarding point 1. above: how can we avoid organisation duplication? If you need to fill in the organisation you work for, ideally it gives you suggestions when you start typing the name of your organisation. So not a long dropdown of all organisations, since users search for different things: long name, short name, abbreviations and there might be typos.

Also: we should look at the possibilities to give users certain rights to be an editor for one project and user for another project. Now you can only have one role and that role is the same for all projects linked to your organisation.

**Comment by KasperBrandt** [comment]
@lauraroverts Yup, I agree on the first point. The 'organisation select box' should be a typeahead (like the one we use in the project editor and for the filters on the listing pages). In case a user's organisation really is not in RSR yet, the user should be able to add a new organisation (similar to project editor again). We should check the name (lowercase) of the organisation before it can be created.

On project specific permissions; that's kind of a hard one. I think it should be possible eventually, but that will require some work, since everything in RSR is currently based on the user permissions to an organisation. I would say it's not in scope of this issue.

**Comment by zzgvh** [comment]
One idea is to presume the email address supplied by the user is OK. We could then trim the workflow to only one back'n'forth:
* User registers and selects an employer
* Employer admin accepts the application
* Email is sent to the user informing of the acceptance and providing a link to close the email loop.

**Comment by nadiagorchakova** [comment]
update: Gabriel says the issue with registering an employment got fixed in the meantime

**Comment by nadiagorchakova** [comment]
was closed during RSR Backlog refinement meeting - the issue is not relevant anymore
